docs/design.md

```diff
@@ -33,143 +31,130 @@ Prometheus writes "blocks" of data, which represent all time series data collect
+Store, rule, and compactor nodes are all expected to scale significantly within a single instance or high availability pair. Similar to Prometheus, functional sharding can be applied for rare cases in which this does not hold true.
 
-TODO(fabxc), TL;DR:
+For example, rule sets can be divided across multiple HA pairs of rule nodes. Store nodes likely are subject to functional sharding regardless by assigning dedicated buckets per region/datecenter.
 
-* Have a hihgly available rule evaluator component which talks to the query layer and writes samples like a regular Prometheus server
-* Treat it like a normal Prometheus server for serving chunks and shipping data to GCS
-* Probably a good idea to define some scope rules, which then allow keeping alerting rules that don't span multiple Prometheus servers on the original collection servers. This retains the reliability model of Prometheus a bit.
+Overall, first-class horizontal sharding is possible but will not be considered for the time being since there's no evidence that it is required in practical setups.
 
 
-### Re-processing old data
+## Cost
 
-TODO(fabxc), TL;DR:
+The only extra cost Thanos adds to an existing Prometheus setup is essentially the price of storing and querying data from the object storage and running of the store node.
 
-* There are many reason to touch data in GCS again: downsampling, deleting low-relevance data/dynamic retention, compacting blocks into larger ones, etc.
-* A simple batch job that downloads, rewrites, and uploads blocks is perfectly sufficient
+Queriers, compactors and rule nodes require as approximately as many compute resources as they save by not doing the same work directly on Prometheus servers.
 
-## Cost estimate
+Data that is just accessed locally in conventional Prometheus setups has to be transferred over the network in Thanos. We generally expect this data shuffling to typically happen in unmetered networks and thus not causing any additional cost.
 
-Suppose the previous example of 750TB of time series data. Particularly query and store instance types and count are a near-blind guess and will likely change.
+Typical object storage prices per GB are at about $0.02. The number of retrievals (typically priced at $0.004 per 10,0000) by the store nodes strongly depend on individual querying pattern. Adding 20% to the total storage cost to account for retrievals and running of store nodes seems like a conservative estimate.
 
-* GCS
-  * storage: $15,000/month
-  * transfer: free within the zone
-  * operations: most ops will be GET ($0.004 per 10k), still unknown but probably minor, assume $200/month for now (500 million ops)
-* HA store node pair
-  * 2 x 64TB provisioned SSD: $21,760
-  * OR 2 x 64TB provisioned HDD: $5120
-  * 2 x n1-standard-32: $1,552
-* Query nodes:
-  * 4 x n1-standard-16: $1,552
+Thereby, storing 100 TB of metric data would cost approximately $2400/month on top of the baseline Prometheus setup.
```

Comment:
Worth to say approximately what does it mean to have 100TB of metrics (with tsdb compression).. We have almost 6 weeks within 170GB (with Prom HA), so we will potentially reach 100TB... with 69 years? xD $2400 looks scarier than it is if someone does not feel how much 100TB really is...

Reply:
I already dialed it down from 750TB before :P But yea, will add some figures.